### ISSUE TITLE
[BUG] Cannot use JDK 16 on Auto-Tune 0.2.4 and 0.3.0-pre-release-3

### ISSUE THREAD

**Issue by Hubok** [Unprotesting/Auto-Tune#25]
**Describe the bug**
Auto-Tune fails to load on startup with a java.lang.reflect.InaccessibleObjectException error if JDK 16 is used.

**To Reproduce**
1. Add either 0.2.4 or 0.3.0-pre-release.3 of Auto-Tune to your server
2. Install JDK 16
3. Attempt to launch and check console output

**Expected behavior**
Auto-Tune to launch properly with JDK 16.

**Screenshots**
```[02:30:36 INFO]: [Auto Tune] Web server has started on port 8124
[02:30:36 WARN]: [[AUTO-TUNE]] [AUTO-TUNE]: Please leave a positive review for Auto-Tune on our spigot page: https://spigotmc.org/resources/auto-tune.85715
[02:30:36 WARN]: [[AUTO-TUNE]] [AUTO-TUNE]: Please donate here if you enjoy the plugin: https://www.patreon.com/Unprotesting
[02:30:36 WARN]: [[AUTO-TUNE]] [AUTO-TUNE]: Using default Auto-Tune 'shops.yml' file
[02:30:36 ERROR]: Error occurred while enabling Auto-Tune v0.13.0-pre-release-3 (Is it up to date?)
java.lang.reflect.InaccessibleObjectException: Unable to make field private final java.util.concurrent.locks.ReentrantLock$Sync java.util.concurrent.locks.ReentrantLock.sync accessible: module java.base does not "opens java.util.concurrent.locks" to unnamed module @7618f22d
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source) ~[?:?]
        at java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source) ~[?:?]
        at java.lang.reflect.Field.checkCanSetAccessible(Unknown Source) ~[?:?]
        at java.lang.reflect.Field.setAccessible(Unknown Source) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerPojo$FieldInfo.<init>(ElsaSerializerPojo.java:247) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerPojo.classInfoDeserialize(ElsaSerializerPojo.java:98) ~[?:?]
        at org.mapdb.DB$classInfoSerializer$1.deserialize(DB.kt:241) ~[?:?]
        at org.mapdb.DB$classInfoSerializer$1.deserialize(DB.kt:231) ~[?:?]
        at org.mapdb.StoreDirectAbstract.deserialize(StoreDirectAbstract.kt:229) ~[?:?]
        at org.mapdb.StoreDirect.get(StoreDirect.kt:546) ~[?:?]
        at org.mapdb.DB.loadClassInfos(DB.kt:331) ~[?:?]
        at org.mapdb.DB.access$loadClassInfos(DB.kt:32) ~[?:?]
        at org.mapdb.DB$elsaSerializer$4.getClassInfo(DB.kt:210) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerPojo.getClassInfo(ElsaSerializerPojo.java:119) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerPojo.deserializeUnknownHeader(ElsaSerializerPojo.java:521) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerBase.deserialize(ElsaSerializerBase.java:1493) ~[?:?]
        at org.mapdb.elsa.ElsaSerializerBase.deserialize(ElsaSerializerBase.java:1479) ~[?:?]
        at org.mapdb.DB$defaultSerializer$1.deserialize(DB.kt:221) ~[?:?]
        at org.mapdb.StoreDirectAbstract.deserialize(StoreDirectAbstract.kt:229) ~[?:?]
        at org.mapdb.StoreDirect.get(StoreDirect.kt:527) ~[?:?]
        at org.mapdb.HTreeMap.valueUnwrap(HTreeMap.kt:1198) ~[?:?]
        at org.mapdb.HTreeMap.getprotected(HTreeMap.kt:644) ~[?:?]
        at org.mapdb.HTreeMap.containsKey(HTreeMap.kt:584) ~[?:?]
        at unprotesting.com.github.Main.loadShopData(Main.java:647) ~[?:?]
        at unprotesting.com.github.Main.onEnable(Main.java:251) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-669]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Paper-669]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Paper-669]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Paper-669]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Paper-669]
        at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:592) ~[patched_1.16.5.jar:git-Paper-669]
        at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:281) ~[patched_1.16.5.jar:git-Paper-669]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1066) ~[patched_1.16.5.jar:git-Paper-669]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:290) ~[patched_1.16.5.jar:git-Paper-669]
        at java.lang.Thread.run(Unknown Source) [?:?]
[02:30:36 INFO]: [[AUTO-TUNE]] Disabling Auto-Tune v0.13.0-pre-release-3
[02:30:37 INFO]: [Auto-Tune] Disabled Version 0.13.0-pre-release-3
```

**Minecraft Server Version**
1.16.5

**Plugin List**
AdvancedNMotd, Auto-Tune, AutoRestart*, ClearLag, DiscordSRV, dynmap*, Dynmap-Towny, Essentials, EssentialsGeoIP, EssentialsSpawn, FastLeafDecay, LogBlock, LuckPerms, PlaceholderAPI, ProtocolLib, QuickShop, Sleep, Towny, TownyChat, Vault, WorldBorder, WorldEdit

**Additional context**
Paper will require JDK 16 for Minecraft 1.17 and warns me to update ASAP. Auto-Tune currently works fine on JDK 11.

**Comment by Dynitios** [comment]
The current version of Lombok is incompatible with JDK16. Updating to 1.18.20 should solve this issue.

**Comment by Hubok** [comment]
I'd be happy to try a patched version of that against 1.16.5 until Paper 1.17 happens.

**Comment by Geolykt** [comment]
If lombok is the issue delombok can be used, otherwise you can always use `illegal-access=permit`

**Comment by Unprotesting** [comment]
This issue is fixed in [0.13.0-Development](https://github.com/Unprotesting/Auto-Tune/tree/0.13.0-Development) as lombok was bumped to `1.18.20` which supports Java 16

**Comment by Unprotesting** [comment]
If this is still an issue please re-open it or use our discord.
